metadata_backend/api/operators.py

```diff
@@ -138,6 +150,26 @@ async def _remove_object_from_db(self,
             LOG.error(reason)
             raise web.HTTPBadRequest(reason=reason)
 
+    async def _insert_new_folder_to_db(self, data: Dict) -> str:
+        """Insert new object folder to database.
+
+        :param data: JSON object containing initial folder info
+        :returns: Folder ID for folder inserted to database
+        """
+        try:
+            data['metadata_objects'] = []
+            insert = await self.db_service.create_folder(data)
+        except (ConnectionFailure, OperationFailure) as error:
+            reason = f"Error happened while getting file: {error}"
+            LOG.error(reason)
+            raise web.HTTPBadRequest(reason=reason)
+        if insert[0]:
```

Comment:
this is somewhat hard to read

Reply:
if you implement create / create_folder (however you want to go) to return id , you can just do following: 
```
id = await self.db_service.create("folder", data)
...
if id: 
   return id
else: 
...
```